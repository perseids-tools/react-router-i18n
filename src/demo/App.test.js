import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

it('renders with full navbar and Home page', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text in another locale', () => {
  const component = renderer.create(<App />);
  const { history } = component.root.findByProps({ href: '/' }).props;

  history.push('/fr/')

  // this is generated randomly; set it to avoid spurious diffs
  history.location.key = 'test';

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
});
