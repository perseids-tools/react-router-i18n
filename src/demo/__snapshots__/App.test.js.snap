// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with full navbar and Home page 1`] = `
<div>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/about"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/topics"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
      >
        Topics
      </a>
    </li>
    <br />
    <li>
      <a
        aria-current={null}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/fr/"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
      >
        Fran√ßais
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/en/"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
      >
        English
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        href="/il/"
        match={
          Object {
            "isExact": true,
            "params": Object {
              "locale": undefined,
            },
            "path": "/:locale(en|fr|il)?",
            "url": "/",
          }
        }
        onClick={[Function]}
      >
        Pig Latin
      </a>
    </li>
  </ul>
  <hr />
  <div>
    <h2>
      Home
    </h2>
    <p>
      This is the homepage
    </p>
  </div>
</div>
`;
